types/twine-sugarcube/macro.d.ts

```diff
@@ -46,6 +46,7 @@ export interface MacroContextObject {
     contents: string;
 }
 
+export type ShadowWrapperCallback<T extends any[]> = (this: MacroContext, ...args: T) => void;
```

Comment:
Does this type need to be explicitly exported?

The project (DefinitelyTyped) seems to prefer the array notation near the paramter, could you change that, pleas?
`type ShadowWrapperCallback<T extends any> = (this: MacroContext, ...args: T[]) => void;`

Reply:
Uh, probably doesn't need to be exported. Will fix. I'm used to languages that don't like the usage of private types in public interfaces.  
  
As for the `T[]` approach, I don't think that works after some testing. For multiple parameters, it makes it think that they are either of the types. (So, `T` becomes `(SomeType | OtherType)` and so doesn't have ordering information as `(SomeType | OtherType)[]`).
![Screenshot_20220102_074240](https://user-images.githubusercontent.com/13157904/147877727-9abe9ad3-4a11-40df-aa3b-999b6ec14282.png)

Though, giving a search in the source code for `T extends any`, I find that there are a number of cases that do what I wrote above.
A few:
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/amazon-connect-streams/index.d.ts#L387
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/check-types/index.d.ts#L78 (kindof)
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/safe-timers/index.d.ts#L20

Reply:
My bad, these are different case. Thank you for explaining!